src/item.cpp

```diff
@@ -2569,6 +2605,22 @@ const std::string &item::symbol() const
     return type->sym;
 }
 
+// Used for craft entity
+static const item *get_most_rotten_component( const item &craft )
+{
+    const item *most_rotten = nullptr;
+    for( const item &it : craft.components ) {
+        if( it.goes_bad() ) {
+            if( most_rotten == nullptr ) {
+                most_rotten = &it;
+            } else if( it.get_relative_rot() > most_rotten->get_relative_rot() ) {
```

Comment:
Since the left and the right legs are the same (static code analyses usually aggro) Can be squashed into ```if( !most_rotten || most_rotten->get_relative_rot() < it.get_relative_rot() )```.

Reply:
Will do, thanks!